fabric-biomes-v1/src/main/java/net/fabricmc/fabric/api/biomes/v1/OverworldBiomes.java

```diff
@@ -107,4 +107,30 @@ public static void addBiomeVariant(Biome replaced, Biome variant, double chance,
 	public static void setRiverBiome(Biome parent, Biome river) {
 		InternalBiomeData.setOverworldRiverBiome(parent, river);
 	}
+
+	/**
+	 * Adds the biome as a large edge biome of the parent biome, with the specified weight.
+	 * Large edges are around the same size as hills biomes.
+	 *
+	 * @param parent the base biome to which the large edge biome is added
+	 * @param edge the biome to be added as a large edge biome
+	 * @param weight the weight of the entry. The weight in this method corresponds to its selection likelihood, with
+	 * heavier biomes being more likely to be selected and lighter biomes being selected with less likelihood.
+	 * @param predicate a predicate of the neighboring biomes and an {@link LayerRandomnessSource} which returns whether the conditions are correct for the large edge biome to spawn.
+	 */
+	public static void addLargeEdgeBiome(Biome parent, Biome edge, double weight, BiomeNeighborsPredicate predicate) {
+		InternalBiomeData.addOverworldLargeEdge(parent, edge, weight, predicate);
+	}
+
+	/**
+	 * Adds a 'sub biome,' similar to minecraft's sunflower plains, that occasionally replaces a portion of the parent biome.
+	 * The sub biome is around the same size as hills biomes.
+	 *
+	 * @param parent the base biome which can have portions replaced by the sub biome
+	 * @param subBiome the biome to be added as a sub biome
+	 * @param chance the chance of replacing the biome with the sub biome
+	 */
+	public static void addSubBiome(Biome parent, Biome subBiome, double chance) {
```

Comment:
We need some way to sort these registrations deterministically so the result doesn't depend on load order. One way could be stable sorting by "parentId;subId;token", where the ids are the biomes' string ids and token is something like a mod id or random token passed to addSubBiome.

Reply:
imo this should be fixed with the current biome api before it's added to my pr